tests/Doctrine/Tests/ORM/Functional/Ticket/GH7854Test.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @group GH7854
```

Comment:
Please add [`@requires PHP >= 7.4`](https://phpunit.readthedocs.io/en/8.4/incomplete-and-skipped-tests.html#skipping-tests-using-requires) to make sure we don't run this test for other PHP versions.

Reply:
@Ocramius do you have any suggestion on how to avoid syntax errors for PHP < 7.3 here? The only thing I can think of is moving the classes to an external file or using `eval()` :nauseated_face:

Reply:
I wouldn't mind bumping the dependency :trollface:

Reply:
I just had a look at [the PHPUnit docs](https://phpunit.readthedocs.io/en/8.4/incomplete-and-skipped-tests.html#skipping-tests-using-requires)

> If you are using syntax that doesn’t compile with a certain PHP Version look into the xml configuration for version dependent includes in The <testsuites> Element

Following the link leads to reading about a way to add a directory to a test suite only when matching some constraints:

```
<testsuites>
  <testsuite name="unit">
    <directory phpVersion="8.0.0" phpVersionOperator=">=">tests/unit</directory>
  </testsuite>
</testsuites>
```

Reply:
I like the first solution better, only one directory to create.

Reply:
Wait the tests pass now, I thought there were an issue with PHP < 7.3 as @lcobucci mentioned. Why is it not the case?

Reply:
Ah that's because the class was defined in the same file at first, isn't it?

Reply:
Yes, then I moved it to a separate file.

Reply:
So I guess we should now decide if we stay with the separate file or if we go with the solution I described above. @lcobucci , @Ocramius , any opinions?

Reply:
Didn't know about that PHPUnit feature :scream: It's much better than having a separate file. @sspat sorry about the back and forth but could you change it? :grimacing:

Reply:
@lcobucci no problem, done!